tcf_website/tests/test_review.py

```diff
@@ -126,3 +129,84 @@ def test_unauthorized_user_delete(self):
         # trying to delete a review you don't have access to should result in
         # 403 - forbidden
         self.assertEqual(response.status_code, 403)
+
+
+class ModelReviewTests(TestCase):
+    """Tests for the Review model."""
+
+    def setUp(self):
+        setup(self)
+
+    def test_count_votes(self):
+        """Test for count votes method"""
+
+        expected_sum = 1
+        self.assertEqual(self.review1.count_votes(), expected_sum)
+
+    def test_count_votes_no_votes(self):
+        """Test for count votes method for when there are no votes"""
+
+        expected_sum = 0
+        self.assertEqual(self.review2.count_votes(), expected_sum)
+
+    def test_upvote(self):
+        """Test for upvote method verify with count_votes"""
+
+        self.review1.upvote(self.user4)
+        expected_sum = 2
+        self.assertEqual(self.review1.count_votes(), expected_sum)
+
+    def test_upvote_already_upvoted(self):
+        """Test for upvote method verify with count_votes"""
+
+        self.review1.upvote(self.user4)
+        self.review1.upvote(self.user4)
+        expected_sum = 1
+        self.assertEqual(self.review1.count_votes(), expected_sum)
+
+    def test_downvote(self):
+        """Test for downvote method verify with count_votes"""
+
+        self.review1.downvote(self.user4)
+        expected_sum = 0
+        self.assertEqual(self.review1.count_votes(), expected_sum)
+
+    def test_upvote_already_downvoted(self):
+        """Test for downvote method verify with count_votes"""
+
+        self.review1.downvote(self.user4)
+        self.review1.downvote(self.user4)
+        expected_sum = 1
+        self.assertEqual(self.review1.count_votes(), expected_sum)
+
+    def test_double_vote(self):
+        """Test for voting twice on same review by same user using vote model"""
+        try:
+            self.assertRaises(IntegrityError, Vote.objects.create(
+                value=-1,
+                user=self.user3,
+                review=self.review1))
+
+        except IntegrityError:
+            pass
```

Comment:
If you have this try/except block because of the error message, you can use `suppress_request_warnings` instead.

Reply:
I tried that and it does not work. I think it is because it isn't a request error it's an actual exception. But now I figured out how to do it without the try except block.